massa-network-exports/src/establisher.rs

```diff
@@ -1,43 +1,43 @@
 // Copyright (c) 2022 MASSA LABS <info@massa.net>
 
-#[cfg(not(test))]
+#[cfg(not(feature = "testing"))]
 use massa_time::MassaTime;
-#[cfg(not(test))]
+#[cfg(not(feature = "testing"))]
 use std::{io, net::SocketAddr};
-#[cfg(not(test))]
+#[cfg(not(feature = "testing"))]
 use tokio::{
     net::{TcpListener, TcpStream},
     time::timeout,
 };
 
-#[cfg(test)]
-pub type ReadHalf = super::tests::mock_establisher::ReadHalf;
-#[cfg(test)]
-pub type WriteHalf = super::tests::mock_establisher::WriteHalf;
-#[cfg(test)]
-pub type Listener = super::tests::mock_establisher::MockListener;
-#[cfg(test)]
-pub type Connector = super::tests::mock_establisher::MockConnector;
-#[cfg(test)]
-pub type Establisher = super::tests::mock_establisher::MockEstablisher;
+#[cfg(feature = "testing")]
+pub type ReadHalf = super::test_exports::mock_establisher::ReadHalf;
+#[cfg(feature = "testing")]
+pub type WriteHalf = super::test_exports::mock_establisher::WriteHalf;
+#[cfg(feature = "testing")]
+pub type Listener = super::test_exports::mock_establisher::MockListener;
+#[cfg(feature = "testing")]
+pub type Connector = super::test_exports::mock_establisher::MockConnector;
```

Comment:
You can remove some of these type

Reply:
In addition, this file is hard to read, I think we can dispatch between testing and not testing with the same technics I used with constants:

It can look like that:

```rust
#[cfg(feature = "testing")]
mod my_types {
    use super::super::test_exports::mock_establisher;
    pub type ReadHalf = mock_establisher::ReadHalf;
    pub type WriteHalf = mock_establisher::WriteHalf;
    pub type Listener = mock_establisher::MockListener;
    pub type Establisher = mock_establisher::MockEstablisher;
}
#[cfg(not(feature = "testing"))]
mod my_types {
    use super::*;
    pub type ReadHalf = tokio::net::tcp::OwnedReadHalf;
    pub type WriteHalf = tokio::net::tcp::OwnedWriteHalf;
    pub type Listener = DefaultListener;
    pub type Establisher = DefaultEstablisher;
}
use my_types::*;
```

And that can be declined for the rest of the file, we can also have an `establisher_testing.rs` (too much ? :rofl: )